### PR TITLE
Fold evaluation program into pyserini directly

### PR DESCRIPTION
**Pull request by MXueguang**
re https://github.com/castorini/pyserini/issues/310

### CODE REVIEW

**Comment by MXueguang** [comment]
output example:
```
Downloading https://raw.githubusercontent.com/castorini/anserini-tools/master/scripts/msmarco/msmarco_doc_eval.py to /home/xueguang/.cache/pyserini/eval/msmarco_doc_eval.py...
/home/xueguang/.cache/pyserini/eval/msmarco_doc_eval.py already exists!
Skipping download.
Running command: ['python3', '/home/xueguang/.cache/pyserini/eval/msmarco_doc_eval.py', '--judgments', 'qrels.msmarco-doc.dev.txt', '--run', 'run.msmarco-doc.doc.txt']
Quantity of Documents ranked for each query is as expected. Evaluating
#####################
MRR @100: 0.2770296928568702
QueriesRanked: 5193
#####################

```

File: pyserini/eval/msmarco_doc_eval.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+import subprocess
+from pyserini.util import download_evaluation_script
+
+parser = argparse.ArgumentParser(description='Evaluate msmarco doc runs.')
+parser.add_argument('--qrel', type=str, required=True, help="Path to qrel file")
+parser.add_argument('--run', type=str, required=True, help="Path to run file")
+args = parser.parse_args()
+
+script_path = download_evaluation_script('msmarco_doc_eval')
+cmd = ['python3', script_path, '--judgments', args.qrel, '--run', args.run]
```
Comment (lintool) [inline pyserini/eval/msmarco_doc_eval.py:11]: I don't think we should try to parse and rename the args. Just pass unchanged to underlying eval script.

**Review by lintool — CHANGES_REQUESTED** [review]
Don't rename args... just pass to underlying eval program. So:

```
java -jar jtreceval-0.0.5-jar-with-dependencies.jar -m map -m P.30 qrels.robust04.txt run.robust04.txt
```
Becomes:

```
python -m pyserini.eval.trec_eval -m map -m P.30 qrels.robust04.txt run.robust04.txt
```

I think this means we'll be mixing `-arg` and `--arg`, but trying to rename the args will be even more confusing, I think...

**Review by lintool — APPROVED** [review]
Pulled branch and verified. Works great!

**Pull request merged**